### PR TITLE
Fix InMemoryVersionStore behaviour on merge

### LINKED ISSUE #1355: InMemoryVersionStore/JGitVersionStore Merge isn't failing when same key was modified on 2 different branches
This code:

``` python
from pynessie import init
from pynessie.model import Put, ContentsKey, IcebergTable
nessie = init()
nessie.create_branch("etl", nessie.get_reference("main").hash_)
key = ContentsKey(["foo", "bar"])
table = IcebergTable("path/to/metadata.json")
table2 = IcebergTable("path/to/another/metadata.json")
op = Put(key, table)
op2 = Put(key, table2)
nessie.commit("etl", nessie.get_reference("etl").hash_, "test", op)
nessie.commit("main", nessie.get_reference("main").hash_, "test", op2)
print(nessie.list_keys("main"))
print(nessie.list_keys("etl"))
nessie.merge("etl", "main", nessie.get_reference("etl").hash_)
print(list(nessie.get_values("main", "foo.bar")))
print(list(nessie.get_values("etl", "foo.bar")))
print([(i.hash_, i.authorTime) for i in nessie.get_log("main")])
print([(i.hash_, i.authorTime) for i in nessie.get_log("etl")])
nessie.delete_branch("etl", nessie.get_reference("etl").hash_)
nessie.delete_branch("main", nessie.get_reference("main").hash_)
nessie.create_branch("main")
```

does not create an exception and instead blindly merges. Maybe a problem w/ python client, maybe a problem w/ Nessie

### PR DESCRIPTION
**Pull request by rymurr**
Fixes #1355

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1356)

### CODE REVIEW

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java [hunk 1]
```diff
@@ -516,6 +516,26 @@ void invalidTags(String invalidTagNameIn, String validHash) {
     );
   }
 
+  @Test
+  void testMerge() throws NessieConflictException, NessieNotFoundException {
```
Comment (snazy) [inline servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java:520]: Can you move this test to `AbstractITVersionStore`?

File: versioned/memory/src/main/java/org/projectnessie/versioned/memory/InMemoryVersionStore.java [hunk 2]
```diff
@@ -315,7 +315,9 @@ public void merge(Hash fromHash, BranchName toBranch, Optional<Hash> expectedBra
       toMerge.add(commit.getValue());
       commit.getValue().getOperations().forEach(op -> keys.add(op.getKey()));
     }
-
+    if (commonAncestor == null) {
```
Comment (snazy) [inline versioned/memory/src/main/java/org/projectnessie/versioned/memory/InMemoryVersionStore.java:318]: Ah! Now `commonAncestor`'s being used ;)

**Review by snazy — COMMENTED** [review]
The change itself looks good, I'd just prefer to have this test in the base test class for all stores.

**Comment by nastra** [comment]
looks like CI is failing with the proposed change :(

**Comment by rymurr** [comment]
Yeah. This isn't the right solution, something deeper is amiss. Will hold off addressing comments until I find out what it is.

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rymurr) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java:520]: done

Comment (rymurr) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/TestRest.java:520]: done

**Pull request closed without merging**